FlowCrypt/Controllers/Threads/ThreadDetailsViewController.swift

```diff
@@ -308,6 +310,43 @@ extension ThreadDetailsViewController {
             updateSpinner(label: "decrypting_title".localized)
         }
     }
+
+    private func handleKeyMismatch(for rawMimeData: Data, at indexPath: IndexPath) {
+        let alert = UIAlertController(
+            title: "error_key_mismatch".localized,
+            message: nil,
+            preferredStyle: .alert
+        )
```

Comment:
We don't actually need the modal at all. I wonder where the original error handling code went? Seems it was somehow refactored away before - I wish we already had ui tests at that time.

Let me take a look if I can find it.

Reply:
Sorry for not being clearer earlier. This is what I want. It's code from 2 years ago, not sure where it got lost meanwhile. I was more sloppy with reviews back then.

```
    private func handleDecryptedMsg(_ decrypted: CoreRes.ParseDecryptMsg) {
        let decryptErrBlocks = decrypted.blocks.filter { $0.decryptErr != nil }
        if let decryptErrBlock = decryptErrBlocks.first {
            let rawMsg = decryptErrBlock.content
            let err = decryptErrBlock.decryptErr?.error
            renderMsgBody(
                "Dould not decrypt message:\n\(err?.type.rawValue ?? "UNKNOWN"): \(err?.message ?? "??")\n\n\n\(rawMsg)",
                color: .red
            )
        } else {
            renderMsgBody(
                decrypted.text,
                color: decrypted.replyType == CoreRes.ReplyType.encrypted ? Constants.green : UIColor.black
            )
        }
        markAsReadIfNotAlreadyMarked()
        hideSpinner()
    }
```

It's this code I want:

```swift
if let decryptErrBlock = decryptErrBlocks.first {
            let rawMsg = decryptErrBlock.content
            let err = decryptErrBlock.decryptErr?.error
            renderMsgBody(
                "Dould not decrypt message:\n\(err?.type.rawValue ?? "UNKNOWN"): \(err?.message ?? "??")\n\n\n\(rawMsg)",
                color: .red
            )
        } 
```

No alert, just render `"Dould not decrypt message:\n\(err?.type.rawValue ?? "UNKNOWN"): \(err?.message ?? "??")\n\n\n\(rawMsg)"` in red color into the text box. Works for all error types. Plus adjust the badges as I mentioned in another comment.

That should handle 99% of cases.